ckanext/ytp/comments/notification_controller.py

```diff
@@ -0,0 +1,73 @@
+import ckan.authz as authz
+import helpers
+import notification_helpers
+import sqlalchemy
+import re
+
+from ckan.lib.base import BaseController, abort
+
+_and_ = sqlalchemy.and_
+
+
+class NotificationController(BaseController):
+
+    def valid_request_and_user(self, thread_or_comment_id):
+        """
+        Check user logged in and perform simple validation of id provided in request
+        :param thread_or_comment_id:
+        :return:
+        """
+        return False if not authz.auth_is_loggedin_user() \
```

Comment:
This could be simplified to return the boolean expression directly:

    return authz.auth_is_loggedin_user() \
        and id \
        and not self.contains_invalid_chars(thread_or_comment_id)